src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -271,8 +271,20 @@ protected override void GetMetadataDependenciesDueToReflectability(ref Dependenc
                 MethodDesc invokeMethod = type.GetMethod("Invoke", null);
                 if (!IsReflectionBlocked(invokeMethod))
                 {
+                    // Size optimization: try to downgrade to metadata-only if possible
+                    bool isActuallyInvokable = !type.IsGenericDefinition;
+                    if (isActuallyInvokable)
+                    {
+                        isActuallyInvokable &= !invokeMethod.Signature.ReturnType.IsByRefLike;
```

Comment:
This won't work once https://github.com/dotnet/runtime/issues/45152 is fixed. POR is to fix that in .NET 8 together with other reflection invoke related functionality and perf issues. cc @steveharter

Reply:
The good news is that the delegate I was after with this (something with a `SpanAction`) seems to have disappeared with the other PRs I did around #83069 so this is no longer needed. It would still be a size savings, not an interesting one.

This was only interesting because the `SpanAction` delegate happened to also have `ValueTuple` in the signature that calls default equality comparer and brings a ton of stuff into hello world.